server/controllers/logstreaming_controller.go

```diff
@@ -117,17 +117,34 @@ func (j *LogStreamingController) GetLogStreamWS(w http.ResponseWriter, r *http.R
 	}
 
 	c, err := j.WebsocketHandler.Upgrade(w, r, nil)
-
 	if err != nil {
 		j.Logger.Warn("Failed to upgrade websocket: %s", err)
 		return
 	}
 
-	defer c.Close()
+	// Add a reader goroutine to listen for socket.close() events.
+	go func() {
+		for {
+			_, _, err := c.ReadMessage()
+			if err != nil {
+				j.Logger.Warn("Failed to read WS message: %s", err)
+				return
+			}
+		}
```

Comment:
Apparently, there needs to be a reader in order to listen for events from the ws client. Since we're using the close handler here: https://github.com/lyft/atlantis/blob/22f56bbee06b1c17c3021d917dd87495b5c2641c/server/controllers/logstreaming_controller.go#L140-L145
Removing the reader makes the closeHandler() non-functional.